File: Problem112/Solution.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func hasPathSum(root *TreeNode, sum int) bool {
	if root == nil || (root.Left == nil && root.Right == nil && sum-root.Val != 0) {
		return false
	}
	return (root.Left == nil && root.Right == nil && sum-root.Val == 0) || hasPathSum(root.Left, sum-root.Val) || hasPathSum(root.Right, sum-root.Val)
}

File: Problem112/Solution.java


import Solution.TreeNode;

public class Solution
{
    public boolean hasPathSum(TreeNode root, int sum)
    {
        if (root == null || (root.left == null && root.right == null && sum - root.val != 0))
        {
            return false;
        }
        return (root.left == null && root.right == null && sum - root.val == 0) || hasPathSum(root.left,
                sum - root.val) || hasPathSum(root.right, sum - root.val);
    }
}